### ISSUE TITLE
Cross-country skiing profile for Android app

### ISSUE THREAD

**Issue by ppete2** [abrensch/brouter#652]
I'm searching for a cross-country profile to be used in Brouter Android app, which is used by apps like Locus.
Would it be possible to develop profile and ship it with a future release of BRouter's android app?

**Comment by poutnikl** [comment]
It would have several aspects:

- enumeration of OSM  way/relation tags relevant to cross-country skiing
- choosing tags that are frequent enough to involving them in the BRouter data files
- using existing tagging for marked cross-country routes
- addressing suitable ways not explicitly marked/tagged as cross-country routes.
- based on above, some profiles may be created

**Comment by limex** [comment]
Plus: You would need 2 profiles: For Classic & Skating

> addressing suitable ways not explicitly marked/tagged as cross-country routes.
Interesting approach. That would be: Take the ski off and hike? :)

**Comment by poutnikl** [comment]
I have meant rather many meadow, field and wood tracks, some of them marked for hiking and/or  biking. They are frequently used during good snow conditions for cross-country skiing too, often with already made ski trails by already passed skiers.

Such tracks are hard to evaluate from few OSM tags.

**Comment by mvscheven** [comment]
Sorry for digging up this old issue. I also would like to have a cross-country profile for brouter and already did a few test with a very simple profile.

One fundamental problem I am facing is, that the key piste:type cannot be used in the profile. 

Using the following as a profile in brouter-web:

---context:way
assign costfactor if piste:type=nordic then 1.0 else 999.01

---context:node
assign initialcost 0

throws an error: `unknown lookup name: piste:type`

Sorry for this maybe stupid question, but I am new to routing profiles and Locus is using exactly this in the build-in SkiingNordic.brf profile.

Thanks for any advice to move forward towards an cross-country skiing profile.

**Comment by limex** [comment]
I also played around with this, but with no success. I get an error. at line 25: unknown lookup value: piste
So far, I didn't have a detailed look at the BRouter code. Maybe one needs to adapt it in order to reference pistes. 

This is my try. Co-writing credits go to Claude. Had to add TXT to the extension to upload it. 
[nordic_skating.brf.txt](https://github.com/user-attachments/files/22735848/nordic_skating.brf.txt)

**Comment by afischerdev** [comment]
@limex 
> unknown lookup value
Yes, this means we have no data about it in the current lookups.dat. This will be included in next version.

And I found more that are not found in lookups.dat.

It is Interesting that there are profiles in the world that won't run with BRouter but comes along with BRouter logic.

Anyway next lookups.dat is not published, so a good time to add new entries. When there is a bigger interest.

I collected some of the possible entries. Please look more closely for values ​​that can be merged (e.g. nordic and skitour) 

piste:type;0000000001 downhill
piste:type;0000000001 nordic
piste:type;0000000001 skitour
piste:type;0000000001 sled
piste:type;0000000001 hiking
piste:type;0000000001 sleigh
piste:type;0000000001 connection

piste:grooming;0000000001 classic
piste:grooming;0000000001 backcountry
piste:grooming;0000000001 skating
piste:grooming;0000000001 scooter
piste:grooming;0000000001 mogul

piste:difficulty;0000000001 novice 
piste:difficulty;0000000001 easy
piste:difficulty;0000000001 intermediate
piste:difficulty;0000000001 advanced
piste:difficulty;0000000001 expert

**Comment by zod** [comment]
I guess those profiles run with BRouter, but with custom lookups.

**Comment by limex** [comment]
@afischerdev thats good news!!

does your value list cover also this grooming combinations? 
piste:grooming=classic;skating
piste:grooming=classic+skating
https://wiki.openstreetmap.org/wiki/Key:piste:grooming#piste:grooming=classic;skating_/_piste:grooming=classic+skating

Regarding Difficulty: 
Advanced and Expert are seperate levels. I suggest also adding freeride, then we have all covered.
https://wiki.openstreetmap.org/wiki/Key:piste:difficulty

**Comment by afischerdev** [comment]
@zod 
Yes, that's what I thought.

@limex 
the other could be added like

piste:grooming;0000000001 classic  skating classic;skating classic+skating

add 

piste:difficulty;0000000001 freeride

**Comment by mvscheven** [comment]
Is there any schedule when the updated lookup.dat will be merged and published?
Thanks for your efforts!
